botfront/imports/api/nlu_model/nlu_model.methods.js

```diff
@@ -165,15 +172,34 @@ if (Meteor.isServer) {
             check(modelId, String);
             check(projectId, String);
             checkIfCan('nlu-admin', getProjectIdFromModelId(modelId));
+            // check the default language of project and the language of model
+            const modelLanguage = NLUModels.findOne({ _id: modelId });
+            const projectDefaultLanguage = Projects.findOne({ _id: projectId });
+            if (modelLanguage.language !== projectDefaultLanguage.defaultLanguage) {
+                try {
+                    NLUModels.remove({ _id: modelId });
+                    Projects.update({ _id: projectId }, { $pull: { nlu_models: modelId } });
+                    return 'Model Deleted';
+                } catch (e) {
+                    throw e;
+                }
+            }
+            return 'Model Not Deleted';
```

Comment:
1) Why does it return a String? Should return a Meteor.Error
2) This message is not helpful
You could `throw Meteor.Error('409', 'The default language cannot be deleted')`

Reply:
Will fix it